dialogue-core/src/main/java/com/palantir/dialogue/core/CautiousIncreaseAggressiveDecreaseConcurrencyLimiter.java

```diff
@@ -130,6 +130,17 @@ void onSuccess(Response result, PermitControl control) {
                 }
             }
 
+            @Override
+            void onFailure(Throwable _throwable, PermitControl control) {
+                control.ignore();
+            }
+        },
+        STICKY() {
```

Comment:
As written I think the queue limit can only ever increase, never drop. I think we should use the HOST_LEVEL behavior here because this is still a per-host queue, just a step more isolated.

This way, at the limit, when a host becomes unhealthy (low limit based on the HOST_LEVEL limiter) the sticky session is also reduced to a similar number of permits and cannot dominate the queue.

Reply:
Have a look at StickyConcurrencyLimitedChannel: it calls Permit#dropped when it fails to schedule on the LimitedChannel it delegates to. That's when it's concurrency limit will drop.

Reply:
Got it, that makes sense!